src/v/cluster/health_monitor_types.cc

```diff
@@ -163,23 +163,33 @@ cluster::node_state adl<cluster::node_state>::from(iobuf_parser& p) {
 
 void adl<cluster::partition_status>::to(
   iobuf& out, cluster::partition_status&& s) {
-    serialize(out, s.current_version, s.id, s.term, s.leader_id);
+    // if revision is not set fallback to old version, we do it here to prevent
+    // old redpanda version from crashing, request handler will decode request
+    // version and base on that handle revision_id field correctly.
+
+    int8_t version = s.revision_id == model::revision_id{}
+                       ? 0
+                       : cluster::partition_status::current_version;
+
+    serialize(out, version, s.id, s.term, s.leader_id, s.revision_id);
```

Comment:
I'm not really sure how this works:

1. the extra revision field is serialized in all cases. So any old code would still crash (or get corrupt data) because it never assumes there is extra data written.
2. in either case, if the new code will in _some_ instance write a version that an older node can't decode what good is it to write any old version?

Reply:
great catch